comictaggerlib/pagelisteditor.py

```diff
@@ -286,6 +288,48 @@ def toggle_double_page(self) -> None:
 
         self.listWidget.setFocus()
 
+    def autodetect_double_pages(self) -> None:
+        if self.comic_archive is None:
+            return
+
+        for page_number in range(self.listWidget.count()):
+            page = self.listWidget.item(page_number)
+            page_dict: ImageMetadata = self.listWidget.item(page_number).data(QtCore.Qt.ItemDataRole.UserRole)[0]
+
+            if "Image" not in page_dict:
+                continue
+
+            image = int(page_dict["Image"])
+
+            # If the page metadata already knows its size, use that. Otherwise, calculate it from the image in the archive
+            if ("ImageWidth" in page_dict) and ("ImageHeight" in page_dict) and ("ImageSize" in page_dict):
+                width = int(page_dict["ImageWidth"])
+                height = int(page_dict["ImageHeight"])
+                size = int(page_dict["ImageSize"])
+            else:
+                width, height, size = self.comic_archive.get_image_dimensions(image)
+
+            if (width <= 0) or (height <= 0) or (size <= 0):
+                continue
+
+            # Is the image at least as wide as tall? It's a double page. Otherwise, it isn't
+            double_page: bool = width >= height
+
+            if double_page:
+                page_dict["DoublePage"] = True
+            elif "DoublePage" in page_dict:
+                del page_dict["DoublePage"]
+
+            # Cache the image size
+            page_dict["ImageWidth"] = str(width)
+            page_dict["ImageHeight"] = str(height)
+            page_dict["ImageSize"] = str(size)
```

Comment:
All of this detection should happen in comicapi and should be included as an optional feature in
`apply_archive_info_to_metadata`
eg
```python
def apply_archive_info_to_metadata(self, md: GenericMetadata, calc_page_sizes: bool = False, detect_double_page: bool = False) -> None:
```